### PR TITLE
catches

### PR DESCRIPTION
**Pull request by ChiefChippy2**
am tired , but catches mistakes and awaits the promise all and stuff. might wanna test

### CODE REVIEW

File: util/command_manager.js [hunk 1]
```diff
@@ -1,34 +1,46 @@
...
-                            console.log(`Deleted ${command.name}`)
+    try{
+        switch(process.argv[2]){
+            case "list":
+                console.log(await CommandList.list(process.env.DISCORD_APPID))
+                break
+            case "push":
+                await resolveAll((await CommandList.parse("commands.js")).map(async (command) => {
+                    const cmd_status = await CommandList.put(process.env.DISCORD_APPID, command)
+                    if(cmd_status.ok){
```
Comment (TiltMeSenpai) [inline util/command_manager.js:58]: This is coming from the Fetch response. If you `await resp.json()`, `cmd_status.ok` doesn't exist here

File: util/command_manager.js [hunk 2]
```diff
@@ -1,34 +1,46 @@
 const CommandList = require("./CommandList");
+function resolveAll(arr){
+    return Promise.all(arr.map(x=>new Promise(r=>x.catch(_=>console.log("Failed doing Something.")).then(r))))
```
Comment (TiltMeSenpai) [inline util/command_manager.js]: Shouldn't you have access to the error somewhere in here? Pretty sure there's something better than "Failed doing Something." we can return.

**Review by TiltMeSenpai — CHANGES_REQUESTED** [review]

File: util/command_manager.js (hunk 2 above)
Comment (ChiefChippy2) [reply util/command_manager.js]: Yes, but it is most probably a fetch error which is multiple lines. Ideally it'd say "failed pushing/deleting (command)" but that needs some refactoring, so I put that as a placeholder

**Comment by ChiefChippy2** [comment]
i just added some stuff for listing commands because it wasn't very nice before.
support for guild commands and choices. TODO (sub)options.
Also fixed https://github.com/TiltMeSenpai/cloudflare-bot/pull/1#discussion_r549888147

**Comment by TiltMeSenpai** [comment]
Merging for now, we should probably separate command syncing utils from bot utils at some point

File: util/command_manager.js (hunk 2 above)
Comment (ChiefChippy2) [reply util/command_manager.js]: > Merging for now, we should probably separate command syncing utils from bot utils at some point

Yeah

**Pull request merged**